### PR TITLE
Use ngsummary in unit testing

### LINKED ISSUE #6650: (Feature Request) Ability to use AOT in unit testing
### Bug Report or Feature Request (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [x] feature request
```

### Desired functionality.

At `@angular/core/testing` v4.2.0, .ngSummary.ts files are introduced to support to use AOT in unit testing ( https://github.com/angular/angular/blob/master/CHANGELOG.md#features-4 ).  Do you have a milestone about this? 
I think it's useful to run unit testing with something like `ng test --aot` .

### PR DESCRIPTION
**Pull request by Quramy**
It closes #6650 .

It allows `ng test --aot`.
If `--aot` option set,  transform `initTestEnvironment` to add the 3rd argument `aotSummaries` (https://angular.io/api/core/testing/TestBed#initTestEnvironment) like replacing `bootstrap` method to use `.ngfactory`

#### limitation
It works only if `angularCompilerPlugin` is supported.

### CODE REVIEW

File: packages/@angular/cli/commands/test.ts [hunk 1]
```diff
@@ -55,6 +55,11 @@ const TestCommand = Command.extend({
         Defaults to the karma config file in .angular-cli.json.`
     },
     {
+      name: 'aot',
```
Comment (lacolaco) [inline packages/@angular/cli/commands/test.ts]: also have to add the same field into `TestOptions` interface

Comment (Quramy) [reply packages/@angular/cli/commands/test.ts]: Thanks for pointing out, I fixed it :)

**Comment by filipesilva** [comment]
@hansl can you check how this fits into the roadmap please?

**Comment by hansl** [comment]
After talking to @tbosch, AOT in unit tests is still experimental. We'll wait until it stabilizes before letting this PR in.

**Comment by Quramy** [comment]
@hansl Thanks for your comment.

Is tbosh currently working on stabilization? If it takes time for that work, I'd like you to consider reviewing and merging.

I implemented this PR's function as opt-in, and it never effect for users who do not use this option.

> AOT in unit tests is still experimental. 

I also understand it. But I have been using AOT unit-test since release angular/core v4.2.1 and I have never encountered any issues about this. And CLI users can not use AOT unit testing without `ng eject` and source code transformation by themselves(it's so painful..), also they are hard to report issues about AOT unit-testing.

**Comment by danwulff** [comment]
Any update on this? I'd love to be able to utilize aot unit testing now that aot compile time is much faster in 5.0

**Comment by JBoothUA** [comment]
I apologize if this is an inappropriate place to post this, but I've been stuck with AoT and Unit testing..

I see here that unit testing with AoT isn't officially supported, but I'm having a fundamental flaw with my source code.

it seems that JiT is required for unit testing and that requires component's templateurls and styleurls to be absolute.  
Also, that AoT is required for our production and that requires component's templateurls and styleurls to be relative to the component.

what is the best practice for templateurls and styleurls if I wish to run AoT and also karma?

**Comment by hansl** [comment]
Closing this as AOT is experimental in unit tests. We are waiting for official word from the compiler team.

**Pull request closed without merging**